lib/tasks/integration.rake

```diff
@@ -0,0 +1,96 @@
+INTEGRATION_TMP_DIR     = Rails.root.join('tmp', 'integration')
+INTEGRATION_FOREMAN_PID = INTEGRATION_TMP_DIR.join('server.pid')
+INTEGRATION_FOREMAN_LOG = Rails.root.join('log', 'integration.log')
+INTEGRATION_PROCFILE    = INTEGRATION_TMP_DIR.join('Procfile')
+RUN_SINGLE_WORKER_BIN   = Rails.root.join('lib', 'workers', 'bin', 'run_single_worker.rb')
+
+namespace :integration do
+  desc "Seed the database and configure assets for integration tests"
+  task :seed => [:env, "test:vmdb:setup", "evm:foreman:seed", :compile_assets]
+
+  desc "Setup the database for integration tests"
+  task :setup => [:db_setup, "evm:foreman:setup"] do
+    mkdir_p INTEGRATION_TMP_DIR
+    rm_f    INTEGRATION_PROCFILE
+    touch   INTEGRATION_PROCFILE
+  end
+
+  desc "Run a foreman server in the background for integration tests"
+  task :run_server => [:setup] do
+    pid = File.read(INTEGRATION_FOREMAN_PID).to_i if File.exist?(INTEGRATION_FOREMAN_PID)
+
+    if pid && (Process.getpgid(pid) rescue nil).present?
+      warn "foreman process already running"
+    else
+      foreman_cmd = %W[
+        foreman start --port=#{ENV['FOREMAN_PORT'] || '3000'}
+                      --root=#{Rails.root}
+                      --procfile=#{INTEGRATION_PROCFILE}
+      ].join(" ")
+
+      pid = Process.spawn(foreman_cmd, [:out, :err] => [INTEGRATION_FOREMAN_LOG, "a"])
+      Process.detach(pid)
+      File.write(INTEGRATION_FOREMAN_PID, pid)
+    end
+  end
+
+  desc "Stop server for integration tests"
+  task :stop_server do
+    if File.exist?(INTEGRATION_FOREMAN_PID)
+      Process.kill("TERM", File.read(INTEGRATION_FOREMAN_PID).to_i)
+      rm INTEGRATION_FOREMAN_PID
+    else
+      warn "PID file for server (#{INTEGRATION_FOREMAN_PID}) doesn't exist... moving on"
+    end
+  end
+
+  desc "With a UI worker for integration tests"
+  task :with_ui => :setup do
+    ui_config = "ui: env PORT=$PORT ruby #{RUN_SINGLE_WORKER_BIN} MiqUiWorker\n"
+    File.write(INTEGRATION_PROCFILE, ui_config, :mode => "a")
+  end
+
+  # Check if a UI worker is running
+  task :ui_ready => :run_server do
+    require 'net/http'
+
+    foreman_port = ENV['FOREMAN_PORT'] || '3000'
+    ui_endpoint  = "http://localhost:#{foreman_port}/api/ping"
+    ui_ready     = false
+
+    15.times do
+      ui_ready = Net::HTTP.get_response(URI(ui_endpoint)).body
+      break if ui_ready == "pong"
```

Comment:
Would it be simpler to just to a HEAD request and check for 200?  I guess it's not really much of a difference, but less is more.

Reply:
I would still want to hit against the `/api/ping` route since it doesn't even load our base controller:

https://github.com/ManageIQ/manageiq-api/blob/fdc94421077ecedfc509c413b17a566d07dbebd0/app/controllers/api/ping_controller.rb#L1-L7

And so that makes the request much cheaper, where a `HEAD /` ends up actually making some database calls.  I don't know if you could do a `HEAD /api/ping`, but seems like it would minimal difference if anything.

Reply:
Sorry, I mean HEAD to /api/ping as opposed to GET /api/ping, but yeah it's going to be a pretty minimal difference.  My only thought was if for some reason we change the pong to something else, then this creates an unnecessary coupling, that a HEAD could avoid.

Reply:
Fair enough.  I can do a quick test to see if we would have to make an addition to that controller to do this.  I honestly don't know the answer off hand, or what it would look like for the code in Ruby (I think `.get_response` is a special method for `net/http` at least).

Reply:
Switched to HEAD in some refactoring commits, and it works.  The `response.code` comes back as a string, which I wasn't expecting, but besides that, it is probably a better approach.  Requires setting up a `http` object ahead of time since there isn't a class method for it, so one extra line of code was required.